### ISSUE TITLE
Table metadata `from_dict` does not take `anonymize_fields`

### ISSUE THREAD

**Issue by simone-mangiante** [sdv-dev/SDV#1321]
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 0.18.0
* Python version: 3.7.12
* Operating System: Debian 10

### Error Description

When instantiating a CopulaGAN model passing the `table_metadata` parameter as a dictionary containing the `anonymize_fields` dict, the `Table` constructor https://github.com/sdv-dev/SDV/blob/e6dfd6b37384f17b593d44d2c96e3d092046b021/sdv/metadata/table.py#L801 does not parse `anonymize_fields` and no Faker provider gets used by the model. 

### Steps to reproduce

I ran this cell from the single table CouplaGAN tutorial notebook, taking as source dataset the Titanic dataset from Kaggle
```
from sdv.tabular import CopulaGAN
import numpy as np

discrete_columns = ["Survived", "Pclass", "Sex", "SibSp", "Parch", "Embarked"]
primary_key = "PassengerId"

# data is a pandas dataframe with the Titanic Kaggle dataset
field_types = {}
for col_type, col_name in zip(data.dtypes, data):
    if col_name == primary_key:
        field_types[col_name] = {"type": "id"}
    else:
        if col_name in discrete_columns:
            field_types[col_name] = {"type": "categorical"}
        else:
            if col_type == np.int64:
                field_types[col_name] = {"type": "numerical", "subtype": "integer"}
            elif col_type == np.float64:
                field_types[col_name] = {"type": "numerical", "subtype": "float"}
            else:
                field_types[col_name] = {"type": "categorical"}

providers = {"Name": "name"}
table_name = "test"

metadata_dict = {'name': table_name}
metadata_dict['fields'] = field_types
metadata_dict['anonymize_fields'] = providers
print(metadata_dict)
print("----------")

model = CopulaGAN(table_metadata=metadata_dict, verbose=True, epochs=10)

print(model._metadata)
print("----------")
print(model._metadata._anonymize_fields)
```
The output is:
```
{'name': 'test', 'fields': {'PassengerId': {'type': 'id'}, 'Survived': {'type': 'categorical'}, 'Pclass': {'type': 'categorical'}, 'Name': {'type': 'categorical'}, 'Sex': {'type': 'categorical'}, 'Age': {'type': 'numerical', 'subtype': 'float'}, 'SibSp': {'type': 'categorical'}, 'Parch': {'type': 'categorical'}, 'Ticket': {'type': 'categorical'}, 'Fare': {'type': 'numerical', 'subtype': 'float'}, 'Cabin': {'type': 'categorical'}, 'Embarked': {'type': 'categorical'}}, 'anonymize_fields': {'Name': 'name'}}
----------
Table(name=test, field_names={'Sex', 'Parch', 'Ticket', 'Survived', 'Pclass', 'SibSp', 'Fare', 'PassengerId', 'Embarked', 'Cabin', 'Age', 'Name'})
----------
{}
```

The `anonymize_fields` dict is empty. This issue is related to #636 

When passing separate fields instead of `table_metadata`, Faker providers get correctly taken by the model. If I execute this cell:
```
from sdv.tabular import CopulaGAN
import numpy as np

discrete_columns = ["Survived", "Pclass", "Sex", "SibSp", "Parch", "Embarked"]
primary_key = "PassengerId"

# data is a pandas dataframe with the Titanic Kaggle dataset
field_types = {}
for col_type, col_name in zip(data.dtypes, data):
    if col_name == primary_key:
        field_types[col_name] = {"type": "id"}
    else:
        if col_name in discrete_columns:
            field_types[col_name] = {"type": "categorical"}
        else:
            if col_type == np.int64:
                field_types[col_name] = {"type": "numerical", "subtype": "integer"}
            elif col_type == np.float64:
                field_types[col_name] = {"type": "numerical", "subtype": "float"}
            else:
                field_types[col_name] = {"type": "categorical"}

providers = {"Name": "name"}
table_name = "test"

metadata_dict = {'name': table_name}
metadata_dict['fields'] = field_types
metadata_dict['anonymize_fields'] = providers
print(metadata_dict)
print("----------")

model = CopulaGAN(field_names=field_types.keys(), field_types=field_types, primary_key=primary_key, anonymize_fields=providers, verbose=True, epochs=10)

print(model._metadata)
print("----------")
print(model._metadata._anonymize_fields)
```

The output is correct:
```
{'name': 'test', 'fields': {'PassengerId': {'type': 'id'}, 'Survived': {'type': 'categorical'}, 'Pclass': {'type': 'categorical'}, 'Name': {'type': 'categorical'}, 'Sex': {'type': 'categorical'}, 'Age': {'type': 'numerical', 'subtype': 'float'}, 'SibSp': {'type': 'categorical'}, 'Parch': {'type': 'categorical'}, 'Ticket': {'type': 'categorical'}, 'Fare': {'type': 'numerical', 'subtype': 'float'}, 'Cabin': {'type': 'categorical'}, 'Embarked': {'type': 'categorical'}}, 'anonymize_fields': {'Name': 'name'}}
----------
Table(name=None, field_names=dict_keys(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']))
----------
{'Name': 'name'}
```

**Comment by npatki** [comment]
Hi @simone-mangiante, thanks for filing this issue.

We are in the process of moving over to an all-new, SDV 1.0 library. The [**SDV 1.0 (Beta)** release](https://docs.sdv.dev/sdv) is already out -- and it contains improvements for creating metadata and anonymizing fields. I believe this issue should be resolved in SDV 1.0. Feel free to test it out and let us know what you think!

**Comment by npatki** [comment]
Hi @simone-mangiante the SDV 1.0 has now been released, so I'm closing the issue as fixed.

Please let us know if you run into any issues with version 1.0. We can always file a new issue or reopen this one to continue the investigation.
